util/zenfs.cc

```diff
@@ -578,21 +600,23 @@ int zenfs_tool_restore() {
 
 int zenfs_tool_dump() {
   Status s;
-  std::unique_ptr<ZonedBlockDevice> zbd = zbd_open(FLAGS_zbd, true, false);
-  if (!zbd) return 1;
-  ZonedBlockDevice *zbdRaw = zbd.get();
-
+  std::unique_ptr<ZonedBlockDevice> zbd;
   std::unique_ptr<ZenFS> zenFS;
-  s = zenfs_mount(zbd, &zenFS, true);
+  std::ostream &json_stream = std::cout;
+
+  s = zenfs_zbd_open(FLAGS_zbd, true, false, zbd);
+  if (!s.ok() || !zbd) return 1;
+
+  json_stream << "{\"zones\":";
```

Comment:
Why did the initial setup of the `json_stream` went before the check of a successful mount?

Reply:
Same as above. We loose the zbd handle when mounting.